packages/stream-model-instance-handler/src/model-instance-document-handler.ts

```diff
@@ -312,4 +312,23 @@ export class ModelInstanceDocumentHandler implements StreamHandler<ModelInstance
       }
     }
   }
+
+  /**
+   *  Helper function to validate if locked fields are being mutated
+   */
+  async _validateLockedFieldsUpdate(model: Model, payload: any): Promise<void> {
+    if ('immutableFields' in model.content) {
```

Comment:
nitpick: you can early return like you do if the array is empty, and save an indentation layer.